src/web/mjs/connectors/ScantradUnion.mjs

```diff
@@ -56,14 +68,17 @@ export default class ScantradUnion extends Connector {
         }));
     }
 
-    _handleConnectorURI( payload ) {
-        /*
-         * TODO: only perform requests when from download manager
-         * or when from browser for preview and selected chapter matches
-         */
-        this.requestOptions.headers.set('x-referer', payload.referer);
-        let promise = super._handleConnectorURI(payload.url);
-        this.requestOptions.headers.delete('x-referer');
-        return promise;
+    async _getMangaFromURI(uri) {
+        let request = new Request(new URL(uri.href), this.requestOptions);
+        let data = await this.fetchDOM(request, 'div.projet-description > h2');
+        return new Manga(this, uri.pathname, this.cleanTitle(data[0].textContent));
+    }
+
+    cleanTitle(title) {
+        if (title.trim().startsWith('[Partenaire] ')) {
```

Comment:
May use replace instead of search and slice

Reply:
Wasn't aware `String.replace` was regex aware. Cool stuff.